### PR TITLE
Compute x ^ n

### PR DESCRIPTION
**Pull request by pradhan1234**
Computing x to the power n in log n time using Divide and Conquer Technique.

### CODE REVIEW

File: 50. Pow(x, n)/Solution.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    
+    // helper method
+    private double myPow(double x, long n) {
+        // case: n = 0
+        if(n == 0) {
+            return 1;
+        }
+        // case: n = 1
+        if(n == 1) {
```
Comment (shivani95thakkar) [inline 50. Pow(x, n)/Solution.java:15]: no need for this condition but it is harmless.

File: 50. Pow(x, n)/Solution.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            return x;
+        }
+        // case: n < 0, find nth power of reciprocal of x
+        // changing type of n to long takes care of cases like when n == Integer.MIN_VALUE
+        // otherwise when sign flips we run into overflow 
+        if(n < 0) {
+            x = 1/x;
```
Comment (shivani95thakkar) [inline 50. Pow(x, n)/Solution.java:22]: This is equivalent to 
 ` if(n<0){
            return 1/pow(x,-n);
        }`

**Review by prit2596 — APPROVED** [review]
Great Work!!

**Comment by pradhan1234** [comment]
LGTM

**Pull request merged**